### PR TITLE
Cleanup extension samples and reset release notes

### LINKED ISSUE #1805: Converge WorkerBindingSamples into core extension examples
The current samples under `samples/Extensions` use very old versions (some major versions old):

https://github.com/Azure/azure-functions-dotnet-worker/blob/887f5a9a62e2255462c968412c2a9d84da03b448/samples/Extensions/Extensions.csproj#L8-L25

At the very least, that should be including:

```xml
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.18.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.13.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.CosmosDB" Version="4.4.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.EventGrid" Version="3.3.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.EventHubs" Version="5.5.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.ServiceBus" Version="5.12.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Storage.Blobs" Version="6.0.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Storage.Queues" Version="5.2.0" />
```

Note that there are two major version changes, but only the Cosmos one should require changes to other parts of the sample.

That said, we should be converging things from `samples/WorkerBindingSamples` into this folder for anything that has been GA'd for that initiative. We just need to ensure that there aren't doc references that will break if we do that. I'm glad to help with that.

### PR DESCRIPTION
**Pull request by liliankasem**
### Issue describing the changes in this PR

resolves #1805

Removing WorkerExtensionBinding sample and updating the Extension sample project.

Future work: Updated best practice for output without using the output binding

### CODE REVIEW

File: samples/Extensions/Blob/BlobFunction.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ namespace SampleApp
 {
     public static class BlobFunction
     {
-        [Function("BlobFunction")]
+        [Function(nameof(BlobFunction))]
```
Comment (SeanFeldman) [inline samples/Extensions/Blob/BlobFunction.cs:11]: ❤️

File: samples/FunctionApp/FunctionApp.csproj [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.ApplicationInsights.WorkerService" Version="2.21.0" />
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.7.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.13.1" />
```
Comment (surgupta-msft) [inline samples/FunctionApp/FunctionApp.csproj:17]: Update this to 1.14.0? - https://www.nuget.org/packages/Microsoft.Azure.Functions.Worker.Sdk/

**Review by surgupta-msft — APPROVED** [review]

**Review by ggailey777 — COMMENTED** [review]
@liliankasem just an FYI we were using some of these code examples in our documentation. The hint is that for most of these I added snippet tags to the source code. This caused a break in our Azure documentation build until they removed the snippets from the source .md files.

Adding @mattchenderson for awareness.

**Comment by liliankasem** [comment]
> @liliankasem just an FYI we were using some of these code examples in our documentation. The hint is that for most of these I added snippet tags to the source code. This caused a break in our Azure documentation build until they removed the snippets from the source .md files.
> 
> Adding @mattchenderson for awareness.

😱 oh no! I didn't realise, thanks for the info - will keep this in mind with future updates. Is there anything we can do to help update the docs?

**Pull request merged**